cognite/client/data_classes/transformations/__init__.py

```diff
@@ -112,8 +116,45 @@ def __init__(
         self.blocked = blocked
         self.schedule = schedule
         self.data_set_id = data_set_id
+        self.source_nonce = source_nonce
+        self.destination_nonce = destination_nonce
         self._cognite_client = cast("CogniteClient", cognite_client)
 
+    def _process_credentials(self, sessions_dict: Dict[str, NonceCredentials], keep_none: bool = False) -> None:
+        def try_get_or_create_nonce(oidc_credentials: Optional[OidcCredentials]) -> Optional[NonceCredentials]:
+            if keep_none and oidc_credentials is None:
+                return None
+
+            key = (
+                f"{oidc_credentials.client_id}:{oidc_credentials.client_secret.__hash__()}"
+                if oidc_credentials
+                else "DEFAULT"
+            )
+
+            ret = sessions_dict.get(key)
+            if not ret:
+                if oidc_credentials and oidc_credentials.client_id and oidc_credentials.client_secret:
+                    credentials = ClientCredentials(oidc_credentials.client_id, oidc_credentials.client_secret)
```

Comment:
don't we have other parameters for client credentials, such as project, scopes etc..?

Reply:
ok this is not transformations credentials type 👍

Reply:
where do we use client credentials when we fail to generate the nonce?

Reply:
on line 150:
if the generation succeeds, OIDC client credentials are set to `None`

if the generation fails, OIDC client credentials are left intact and sent to the api

Reply:
I seee thanks